### DISCUSSION TITLE
Use Hasura within Docker

### DISCUSSION THREAD

**Discussion by toton6868** [hasura/graphql-engine#7114] [General]
I am developing an automated solution where I am using an Architecture that is something similar to follows.

1. Building custom PostgreSQL Docker
2. Add 2-3 more tools within the docker
3. Want to use Hasura within the docker using that existing docker PostgreSQL

Is it possible? Where I am getting issues is the DIND (Docker within docker) part.

**Comment by Xetera** [comment]
Just out of curiosity, is there a reason why you're trying to bundle 2 independent processes inside the same container? I imagine you would quickly run into coupling problems when it comes to scaling, sharing resources and failures

**Comment by toton6868** [comment]
I am also doing this out of curiosity. Trying to do some R&D with docker. Trying to merging a few processes together inside a single docker and want to measure its performances.

**Comment by webdeb** [comment]
I strongly feel, that this issuequeue becomes a spam board of a lot of unrelated stuff.
@rikinsk what does it have to do with deployment-cloud when the OP is asking to help with his docker R&D to run two containers inside one.

---

Despite the fact, that your question has nothing to do with hasura, it's still interesting

**Yes you can totally do it**

Just use the official `docker:dind` container, and start your independent `postgres` and `hasura` containers within it. you will have to use **docker-compose** anyway.

You could also experiment with the depth, why not?

```
docker 
   --> docker
    --> docker
      --> docker
        ...
         --> docker compose with hasura & postgres
```

Don't forget to pass the ports through ;)
And when you have some results, I'd love to check it out, maybe a cool blog-post or something with some metrics and numbers. Would be interesting.
